src/languages/css.js

```diff
@@ -21,14 +21,15 @@ export default function(hljs) {
           hljs.APOS_STRING_MODE,
           hljs.QUOTE_STRING_MODE,
           hljs.CSS_NUMBER_MODE,
+          FUNCTION_LIKE
```

Comment:
This can't work because FUNCTION_LIKE doesn't exist yet, not until after it's created - it can't refer to itself.  So you first have to create it, then add it to the inside `contains`... but I'm wondering why this is necessary and if we couldn't just collapse the rule and remove `returnBegin`... then you'd just use `contains: "self"` as the rule would self-nest.

It seems wrong to match EVERY possible function as a built_in (feel like we've had this discussion before)... if we just matched actual built-ins you could do this with `beginKeyword` instead... and simplify as I mentioned above.  If not then as I said you'll need for first create the mode, then modify it to reference itself.

Reply:
IE, something like:

```js
const MODE = { ... } // create it
MODE.contains[1].contains.push(MODE)  // modify it to point to itself
```

It's definitely worth considering if we could simplify the rule though.

Reply:
Another question worth asking is whether we 100% need to know we're inside a function.. if not you may not need the nesting at all... just match `var(`, color var... match `foo(`, color foo... There might be no need to open a mode, look for the final `)`, etc...

Reply:
> just match `var(`, color `var`... match `foo(`, color `foo`, etc...

Looks simpler this way 👍